text/0049-network-ids.md

```diff
@@ -0,0 +1,191 @@
+  to add visual noise or untranslated strings in production -- it may be fine to display nothing in prod as a way of distinguishing it, depending on the application.
+* Take advantage of the `expected_network_id` parameter to help prevent mistakes
+
+When the client is a cross-chain bridge, configuration should be organized so that network-id is determined at the same time that the network-id for the other chain is determined,
+and there should be a single source of truth for this association. It should be made very hard to accidentally configure the bridge to talk to e.g. an Ethereum testnet and also Mobilecoin Prod.
+
+For example, either a hard-coded table or a configuration toml file could be created for the bridge which associates to a Mobilecoin network id:
+
+* URLs of mobilecoin services
+* URLs of any other chain services
+* Network ids of any other chain networks
+* Contract ids for any relevant contracts
+
+This is better than each of these being independent configuration parameters to the software. They should all be changing in lock-step.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This adds more configuration to the network.
+
+Clients may send a few more bytes with each request.
+
+This proposal does NOT, at this time, propose to hash the network id into the responder id. While that would logical, it is also an enclave change.
+
+This means that if there is a configuration problem, it's possible that not all nodes in the network have the same network id, but in this case,
+we can fall-back to the attestation trust roots to prevent a problem, and no harm was done relative to status quo. Eventually users who connect
+to the misconfigured servers will be unable to make progress and report an error, and the configuration can be fixed.
+
+This does not, at this time, propose to make the same API change in the Fog servers. Putting this in the Fog servers would mean that balance checks
+cannot succeed without getting this error if there is mixing of test and prod, so Fog clients would fail faster in case of mixing networks.
+But just putting it in consensus seems higher priority because that prevents loss of funds, and minting problems.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The primary alternative is, do nothing:
+
+* Continue to use ad-hoc configuration differences like attestation trust roots and token ids as the primary means to distinguish dev, test, prod environments.
+* Continue to introduce more such ad-hoc configuration differences as more features are added.
+* Cross-chain bridges primarily lean on attestation roots to avoid e.g. accidentally minting on the wrong network, but it is a bit harder to manage the configuration,
+  because attestation roots live in files.
+
+Another alternative: The Ethereum [JSON-RPC API](https://ethereum.org/en/developers/docs/apis/json-rpc/) returns the network id as a string, but in practice all the ids are actually integers.
```

Comment:
Do we have any info on why they ended up sticking with integers?
My main thought is to reduce usage errors i.e. `my-network` vs `mynetwork`, I'm guessing we aren't concerned with that use/misuse.

Reply:
that is a good point

would be pretty lame if there were like "Mobi1eCoin" net or something